app/models/shipit/stack.rb

```diff
@@ -86,7 +86,7 @@ def lock_author=(user)
     after_commit :sync_github_if_necessary, on: :update
 
     def sync_github_if_necessary
-      if archived_since_previously_changed? && archived_since.nil?
+      if (archived_since_previously_changed? && archived_since.nil?) || branch_previously_changed?
```

Comment:
Since this method is called on update, is `*_previously_changed?` necessary?

Reply:
`on :update` runs on any change. Checking for `branch_previously_changed?` ensures we don't sync on _all_ updates.